### PR TITLE
PYTHON-3562 Type annotation of `bson.json_utils.loads` is incorrect (only accepts `str`)

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Could we add a test for bytes if we don't have one already?

**Comment by juliusgeo** [comment]
I added a super simple test that tests that we can load these types.

File: test/test_json_util.py [hunk 1]
```diff
@@ -71,6 +71,11 @@ def round_trip(self, doc, **kwargs):
     def test_basic(self):
         self.round_trip({"hello": "world"})
 
+    def test_loads_bytes(self):
+        string = rb'{"hello": "world"}'
+        for type in [bytes, bytearray]:
+            json_util.loads(type(string))
```
Comment (ShaneHarvey) [inline test/test_json_util.py:77]: In this case using a list makes the code less understandable from a type perspective. I suggest:
```
            json_util.loads(b'{"hello": "world"}')
            json_util.loads(bytearrary(b'{"hello": "world"}'))
```

Comment (ShaneHarvey) [reply test/test_json_util.py:77]: (with assertions that the loaded value is as expected).

Comment (juliusgeo) [reply test/test_json_util.py]: Done.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**